### ISSUE TITLE
Multiple planes with Flyback

### ISSUE THREAD

**Issue by camille-lab** [MouseLand/suite2p#487]
Hi all,
First of all, many thanks for this fantastic work!

I am doing multiple planes imaging using a piezo and get a blank flyback frame.
ScanImage saves the N relevant planes + 1 flyback plane in a file.
When running suite2p, I set the number of planes to N+1 and it works just fine.

It would be nice however to tell suite2p that my last plane is a flyback and I do not wish to register or detect ROIs from that plane.
In other words, it would save time if you could add a "flyback" field in the GUI that would skip the processing of the last plane (at least in ScanImage-generated files)

Thanks!
Camille

**Comment by nickdelgrosso** [comment]
Hi Camille,

Thank you for this feature request!  Agreed, being able to exclude planes from ROI detection would be very useful.  We'll discuss this feature and keep this issue updated.  

Best wishes,

Nick

**Comment by nickdelgrosso** [comment]
@camille-lab , would it work to exclude the flyback plane entirely from the Suite2p processing (registration + ROI detection + activity extraction), or is this specific to ROI detection?

**Comment by camille-lab** [comment]
Yes, excluding it completely would work!
Thanks

**Comment by asumser** [comment]
Hi all,
first, thank you for the fantastic software!
I am seconding that an optional exclusion of the flyback frame would be very useful. Especially since recently (currently running v0.9.2) suite2p gives an error message if no ROIs are found on the flyback frame (sparse mode), leading to an interruption of the batch and it won't create the combined folder. Only workaround I came up with is to reduce threshold_scaling and increase max_iterations, meaning that I have to do more manual curating.

>>>ERROR<<<
Traceback (most recent call last):
  File "c:\programdata\anaconda3\envs\suite2p\lib\runpy.py", line 193, in _run_module_as_main
>>>ERROR<<<
    "__main__", mod_spec)
  File "c:\programdata\anaconda3\envs\suite2p\lib\runpy.py", line 85, in _run_code
>>>ERROR<<<
    exec(code, run_globals)
  File "c:\programdata\anaconda3\envs\suite2p\lib\site-packages\suite2p\__main__.py", line 78, in <module>
>>>ERROR<<<
    main()
  File "c:\programdata\anaconda3\envs\suite2p\lib\site-packages\suite2p\__main__.py", line 71, in main
    run_s2p(ops, db)
  File "c:\programdata\anaconda3\envs\suite2p\lib\site-packages\suite2p\run_s2p.py", line 397, in run_s2p
>>>ERROR<<<
    op = run_plane(op, ops_path=ops_path)
  File "c:\programdata\anaconda3\envs\suite2p\lib\site-packages\suite2p\run_s2p.py", line 224, in run_plane
    ops, stat = extraction.extract(ops, cell_pix, cell_masks, neuropil_masks, stat)
  File "c:\programdata\anaconda3\envs\suite2p\lib\site-packages\suite2p\extraction\extract.py", line 161, in extract
>>>ERROR<<<
    raise ValueError("stat array should not be of length 0 (no ROIs were found)")
ValueError: stat array should not be of length 0 (no ROIs were found)


Thanks,
Anton

**Comment by carsen-stringer** [comment]
I've added this to the github version. See the tooltip in the GUI over "ignore_flyback" for more info, in the GUI put in a list of ignored planes (0-based) "0,10" or in a notebook provide a list of planes e.g. [0, 10]

let me know if it doesn't work, closing for now

**Comment by maristanye** [comment]
Hi,

I do not quite get the explanation in suite2p on how to exlcude flyback frames. What would be the input in ignore_flyback if I want always to exclude the 3 plane? Is [2,0] or [1,1,0]?
